src/general/MainMenu.cs

```diff
@@ -752,6 +790,33 @@ private void WarnAboutNoEnabledMods()
         }
     }
 
+    private float TrackMenuPerformance(float delta)
+    {
+        var currentFrameRate = 1 / delta;
+
+        // If this is the first frame, do not use the average of the frame delta and 0
+        if (averageFrameRate == 0)
+            return currentFrameRate;
+
+        // Not an exact average by any means, but good enough for this purpose
+        return (averageFrameRate + currentFrameRate) / 2;
+    }
+
+    private void WarnAboutLowPerformance()
+    {
+        if (averageFrameRate < 30 && secondsInMenu >= 15)
+        {
+            GD.Print($"Average frame rate is {averageFrameRate}, prompting to disable 3D backgrounds");
+            lowPerformanceWarning.PopupIfNotDismissed();
+            canShowLowPerformanceWarning = false;
+        }
+    }
+
+    private void OnLowPerformanceDialogConfirmed()
+    {
+        Settings.Instance.UpdateSettingExternally(Settings.Instance.Menu3DBackgroundEnabled, false);
```

Comment:
The new settings should be saved. Otherwise the problem appears again after restarting the game.

Also did I notice it or how is the remembering if the notice shouldn't be shown again checkbox is handled?

Reply:
> how is the remembering if the notice shouldn't be shown again checkbox is handled?

The popup uses the already in-use PermanentlyDismissibleDialog. It seems to work on my end; just needed to be added to DismissibleNotice.cs.

Reply:
With that class you should call `PopupIfNotDismissed` to show it, I don't remember seeing you call that method but I could be wrong. Also in this situation it should act opposite, so if you dismiss the dialog then the "don't show again" state should be remembered, right now PermanentlyDismissibleDialog is setup to only remember the permanent dismiss on confirm. So you need to actually add a new export variable to that class that makes it remember the dismiss state when it is dismissed.